Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py

```diff
@@ -1813,6 +1813,195 @@ def test_module(client: Client, *_) -> Tuple[str, Dict[Any, Any], Dict[Any, Any]
+
+    :type client: ``Client``
+    :param Client: XSOAR client to use
+
+    :type args: ``Dict[str, Any]``
+    :param args:
+        all command arguments, usually passed from ``demisto.args()``.
+        ``args['id']`` incident id to retrieve
+        ``args['lastUpdate']`` when was the last time we retrieved data
+
+    :return:
+        A ``List[Dict[str, Any]]`` first entry is the incident (which can be completely empty) and others are the new entries
+
+    :rtype: ``List[Dict[str, Any]]``
+    """
+
+    ticket_id = args.get('id')
+    demisto.debug(f'Getting update for remote {ticket_id}')
+    last_update = arg_to_timestamp(
+        arg=args.get('lastUpdate'),
+        arg_name='lastUpdate',
+        required=True
+    )
+    demisto.debug(f'last_update is {last_update}')
+
+    ticket_type = client.get_table_name(str(args.get('ticket_type', '')))
+    number = str(args.get('number', ''))
+    get_attachments = args.get('get_attachments', 'false')
+    custom_fields = split_fields(str(args.get('custom_fields', '')))
+    additional_fields = argToList(str(args.get('additional_fields', '')))
+
+    result = client.get(ticket_type, ticket_id, generate_body({}, custom_fields), number)
+    if not result or 'result' not in result:
+        return 'Ticket was not found.'
+
+    if isinstance(result['result'], list):
+        if len(result['result']) == 0:
+            return 'Ticket was not found.'
+        ticket = result['result'][0]
+    else:
+        ticket = result['result']
+
+    ticket_last_update = arg_to_timestamp(
+        arg=ticket.get('sys_updated_on'),
+        arg_name='sys_updated_on',
+        required=False
+    )
+
+    demisto.debug(f'ticket_last_update is {ticket_last_update}')
+
+    if last_update > ticket_last_update:
+        demisto.debug(f'Nothing new in the ticket')
+        ticket = {}
+    else:
+        demisto.debug(f'ticket is updated: {ticket}')
+
+    # get latest comments
+    entries = []
+
+    sys_param_limit = args.get('limit', client.sys_param_limit)
+    sys_param_offset = args.get('offset', client.sys_param_offset)
+
+    sys_param_query = f'element_id={ticket_id}^element=comments^ORelement=work_notes'
+
+    comments_result = client.query('sys_journal_field', sys_param_limit, sys_param_offset, sys_param_query)
+    demisto.debug(f'Comments result is {comments_result}')
+
+    if not comments_result or 'result' not in comments_result:
+        demisto.debug(f'Pull result is {ticket}')
+        return [ticket]
+
+    for note in comments_result.get('result'):
+        entry_time = arg_to_timestamp(
+            arg=note.get('sys_created_on'),
+            arg_name='sys_created_on',
+            required=False
+        )
+        demisto.debug(f'entry_time is {entry_time}')
+
+        if last_update > entry_time:
+            continue
+
+        entries.append({
+            'Type': 1,
+            'Category': 'chat',
+            'ContentsFormat': 'text',
```

Comment:
use the constants, this is prone to errors

Reply:
not rony's comments here